File: backend/src/main/java/com/github/aadvorak/artilleryonline/battle/calculations/ForceCalculator.java
package com.github.aadvorak.artilleryonline.battle.calculations;

import com.github.aadvorak.artilleryonline.battle.model.BattleModel;

public interface ForceCalculator<C extends BodyCalculations> {
    ForceAtPoint calculate(C calculations, BattleModel battleModel);
}

File: frontend/playground/battle/calculator/force-calculator.ts
import type {BodyForce} from "~/playground/battle/calculator/body-force";
import type {BodyCalculations} from "~/playground/data/calculations";
import type {BattleModel} from "~/playground/data/model";

export interface ForceCalculator<C extends BodyCalculations> {
  calculate: (calculations: C, battleModel: BattleModel) => BodyForce[]
}
